worldedit-core/src/main/java/com/sk89q/worldedit/util/collection/LocatedBlockList.java

```diff
@@ -19,74 +19,101 @@
 
 package com.sk89q.worldedit.util.collection;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
-import com.google.common.collect.ImmutableList;
+import com.google.common.collect.AbstractIterator;
 import com.sk89q.worldedit.math.BlockVector3;
 import com.sk89q.worldedit.util.LocatedBlock;
 import com.sk89q.worldedit.world.block.BaseBlock;
 import com.sk89q.worldedit.world.block.BlockStateHolder;
+import it.unimi.dsi.fastutil.longs.LongArrayList;
+import it.unimi.dsi.fastutil.longs.LongListIterator;
 
 import javax.annotation.Nullable;
-import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Collections;
 import java.util.Iterator;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.ListIterator;
-import java.util.Map;
+import java.util.function.Predicate;
+import java.util.function.ToLongFunction;
+
+import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
  * Wrapper around a list of blocks located in the world.
  */
 public class LocatedBlockList implements Iterable<LocatedBlock> {
 
-    private final Map<BlockVector3, LocatedBlock> map = new LinkedHashMap<>();
+    private final BlockMap blocks = BlockMap.create();
+    private final LongArrayList order = new LongArrayList();
```

Comment:
While some parts of the new code (`BlockMap`) appear to take possibility of taller worlds into account, this one doesn't. Is this something that will continue to work and this will be changed later, or is this going to limit world height to +-2048 blocks?

Reply:
There will be a config option to transform this from a `0-4095` range to normal integer limits, at the price of more memory. 

Also, if you think that storing the Y as signed would be better than unsigned, I can do that instead. I just assumed unsigned to begin with.

Reply:
Vanilla `BlockPos.toLong` is signed so I assumed this is the same way here. But it doesn't matter too much.